youtube_dl/extractor/deezer.py

```diff
@@ -45,38 +32,48 @@ def _real_extract(self, url):
              r'naboo\.display\(\'[^\']+\',\s*(.*?)\);\n'),
             webpage, 'data JSON')
         data = json.loads(data_json)
+        return id, webpage, data
+
+
+class DeezerPlaylistIE(DeezerBaseInfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?deezer\.com/(../)?playlist/(?P<id>[0-9]+)'
+    _TEST = {
+        'url': 'http://www.deezer.com/playlist/176747451',
+        'info_dict': {
+            'id': '176747451',
+            'title': 'Best!',
+            'uploader': 'anonymous',
+            'thumbnail': r're:^https?://(e-)?cdns-images\.dzcdn\.net/images/cover/.*\.jpg$',
+        },
+        'playlist_count': 29,
+    }
 
-        playlist_title = data.get('DATA', {}).get('TITLE')
-        playlist_uploader = data.get('DATA', {}).get('PARENT_USERNAME')
+    def _real_extract(self, url):
+        playlist_id, webpage, data = self.get_data(url)
+
+        playlist_title = data.get('DATA').get('TITLE')
+        playlist_uploader = data.get('DATA').get('PARENT_USERNAME')
         playlist_thumbnail = self._search_regex(
             r'<img id="naboo_playlist_image".*?src="([^"]+)"', webpage,
             'playlist thumbnail')
 
-        preview_pattern = self._search_regex(
-            r"var SOUND_PREVIEW_GATEWAY\s*=\s*'([^']+)';", webpage,
-            'preview URL pattern', fatal=False)
         entries = []
-        for s in data['SONGS']['data']:
-            puid = s['MD5_ORIGIN']
-            preview_video_url = preview_pattern.\
-                replace('{0}', puid[0]).\
-                replace('{1}', puid).\
-                replace('{2}', s['MEDIA_VERSION'])
+        for s in data.get('SONGS').get('data'):
             formats = [{
                 'format_id': 'preview',
-                'url': preview_video_url,
+                'url': s.get('MEDIA')[0].get('HREF'),
```

Comment:
Same.

Reply:
I've done `s.get('MEDIA', [{}])[0].get('HREF')` so that if there is no key `MEDIA`, there is no error on `[0]`.